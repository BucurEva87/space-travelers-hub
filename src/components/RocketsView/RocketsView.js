import React from 'react';
import { useSelector } from 'react-redux';
import Rocket from '../Rocket/Rocket';
import style from './RocketsView.module.scss';

const RocketView = () => {
  const rockets = useSelector((state) => state.rockets);

  return (
    <section className={style['rockets-container']}>
      <ul className={style['rockets-list']}>
        { rockets.list.length ? rockets.list.map((rocket) => {
          const {
            id, name, description, flickrImages, reserved,
          } = rocket;

          return (
            <li key={id}>
              <Rocket
                id={id}
                name={name}
                description={description}
                flickrImages={flickrImages[0]}
                reserved={reserved}
              />
            </li>
          );
        }) : (
          <div className={style.loader}>
            <img src={`${process.env.PUBLIC_URL}images/spinner.gif`} alt="Spinning loader" />
            <p>Data is currently loading. Please wait...</p>
          </div>
        ) }
      </ul>
    </section>
  );
};

export default RocketView;
