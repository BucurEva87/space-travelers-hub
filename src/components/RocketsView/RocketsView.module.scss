.rockets-container {
  width: 100%;
}

.rockets-list {
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 2rem;
}

.loader {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 100%;
}
