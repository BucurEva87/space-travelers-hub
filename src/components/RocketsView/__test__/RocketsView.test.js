import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import RocketsView from '../RocketsView';

let getByTestId;
let getByRole;

beforeEach(() => {
  const component = render(
    <Provider store={store}>
      <RocketsView />
    </Provider>,
  );

  getByTestId = component.getByTestId;
  getByRole = component.getByRole;
});

describe('RocketsView component', () => {
  test('renders on the screen', () => {
    expect(getByTestId('rockets-container')).toBeInTheDocument();
  });

  test('displays a loading spinner while fetching the data from the API', () => {
    expect(getByRole('img').src.split('/').reverse()[0]).toBe('spinner.gif');
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <RocketsView />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
