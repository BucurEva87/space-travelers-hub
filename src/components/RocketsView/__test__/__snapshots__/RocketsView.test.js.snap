// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketsView component matches snapshot 1`] = `
<section
  className="rockets-container"
  data-testid="rockets-container"
>
  <ul
    className="rockets-list"
  >
    <div
      className="loader"
    >
      <img
        alt="Spinning loader"
        src="images/spinner.gif"
      />
      <p>
        Data is currently loading. Please wait...
      </p>
    </div>
  </ul>
</section>
`;
